cardano-node/src/Cardano/Node/Configuration/TopologyP2P.hs

```diff
@@ -264,3 +297,25 @@ readTopologyFileOrError tr nc =
   >>= either (\err -> error $ "Cardano.Node.Configuration.TopologyP2P.readTopologyFile: "
                            <> Text.unpack err)
              pure
+
+--
+-- Checking for chance of progress in bootstrap phase
+--
+
+-- | This function returns false if non-trustable peers are configured
+--
+hasTrustablePeers :: NetworkTopology -> Bool
+hasTrustablePeers (RealNodeTopology (LocalRootPeersGroups lprgs) _ _ ubp) =
+    case ubp of
+      DontUseBootstrapPeers   -> True
```

Comment:
I don't have a better name than `hasTrustablePeers`, but this name is a bit misleading if `DontUseBootstrapPeers` returns `True`.

Reply:
Good point! I'll think of something different!